internals/cli/ui/io_unix.go

```diff
@@ -2,23 +2,80 @@
 
 package ui
 
-import "os"
+import (
+	"io"
+	"os"
+)
 
-// NewUserIO creates a new UserIO middleware from os.Stdin and os.Stdout and adds tty if it is available.
-func NewUserIO() UserIO {
+// ttyIO is the implementation of the IO interface that can use a TTY.
+type ttyIO struct {
+	input  *os.File
+	output *os.File
+	tty    *os.File
+}
+
+// NewUserIO creates a new ttyIO if a TTY is available, otherwise it returns a standardIO.
+func NewUserIO() IO {
 	tty, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
 	if err == nil {
-		return UserIO{
-			Input:        os.Stdin,
-			Output:       os.Stdout,
-			tty:          tty,
-			ttyAvailable: true,
+		return ttyIO{
+			input:  os.Stdin,
+			output: os.Stdout,
+			tty:    tty,
 		}
 	}
 
-	return NewStdUserIO()
+	return newStdUserIO()
+}
+
+// Prompts simply returns Stdin and Stdout, when both input and output are
+// not piped. When either input or output is piped, Prompts attempts to
+// bypass stdin and stdout by connecting to /dev/tty on Unix systems when
+// available. On systems where tty is not available and when either input
+// or output is piped, prompting is not possible so an error is returned.
```

Comment:
This comment seems outdated and/or may need to be moved (partially) to a different place in the code that is now responsible for this.